code/go/0chain.net/smartcontract/storagesc/challenge.go

```diff
@@ -152,8 +152,16 @@ func (sc *StorageSmartContract) blobberReward(t *transaction.Transaction,
 		if err != nil {
 			return fmt.Errorf("moving partial challenge to write pool: %v", err)
 		}
-		alloc.MovedBack += back
-		blobAlloc.Returned += back
+		alloc.MovedBack, err = currency.AddCoin(alloc.MovedBack, back)
```

Comment:
We should avoid return value to `alloc.MoveBack` directly, as if the `AddCoin()` failed, the `alloc.MovedBack` will be set to 0, which is not what we want.